presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/api/PackageAPI.java

```diff
@@ -141,7 +147,7 @@ public void run()
                 command.runCommand();
             }
             catch (Exception e) {
-                // TODO: Add to logger
+                LOGGER.error("Encountered exception", e);
```

Comment:
I would add an additional `catch` before this one, catching `PrestoManagerException` specifically, and logging with `error(e.getMessage(), e.getCause())`. Then the message from the exception is logged, and the exception it wraps (if there is one) will be logged (if there isn't one, nothing happens, at least when I tested it).

Then the current `catch` could log `error("Unhandled exception", e)`.